pkg/authorization/maximal_permission_policy_authorizer.go

```diff
@@ -79,55 +74,34 @@ func NewMaximalPermissionPolicyAuthorizer(kubeInformers kcpkubernetesinformers.S
 			)
 		},
 		delegate: delegate,
-	}, nil
+	}
 }
 
 type MaximalPermissionPolicyAuthorizer struct {
-	delegate authorizer.Authorizer
-
 	getAPIBindingReferenceForAttributes func(attr authorizer.Attributes, clusterName logicalcluster.Name) (ref *apisv1alpha1.ExportReference, found bool, err error)
 	getAPIExportByReference             func(exportRef *apisv1alpha1.ExportReference) (ref *apisv1alpha1.APIExport, found bool, err error)
 	newAuthorizer                       func(clusterName logicalcluster.Name) authorizer.Authorizer
+	delegate                            authorizer.Authorizer
 }
 
 func (a *MaximalPermissionPolicyAuthorizer) Authorize(ctx context.Context, attr authorizer.Attributes) (authorizer.Decision, string, error) {
 	// get the cluster from the ctx.
 	lcluster, err := genericapirequest.ClusterNameFrom(ctx)
 	if err != nil {
-		kaudit.AddAuditAnnotations(
-			ctx,
-			MaximalPermissionPolicyAuditDecision, DecisionNoOpinion,
-			MaximalPermissionPolicyAuditReason, fmt.Sprintf("error getting cluster from request: %v", err),
-		)
 		return authorizer.DecisionNoOpinion, MaximalPermissionPolicyAccessNotPermittedReason, err
 	}
 
 	bindingLogicalCluster, bound, err := a.getAPIBindingReferenceForAttributes(attr, lcluster)
 	if err != nil {
-		kaudit.AddAuditAnnotations(
-			ctx,
-			MaximalPermissionPolicyAuditDecision, DecisionNoOpinion,
-			MaximalPermissionPolicyAuditReason, fmt.Sprintf("error getting API binding reference: %v", err),
```

Comment:
Not sure if this matters, but it looks like we're losing small bits of info (e.g.: error getting API binding reference) with these changes. I think it's ok though?

Reply:
In case of error, this information will definitely land in the audit logs before anonymization, we're not losing it 👍 I could `fmt.Errorf("error getting API binding reference: %w", err)` below.

Reply:
Right, I am referring to some of the `fmt.Sprintf` content being lost in this PR

Reply:
let me actually do that before we merge.

Reply:
Thanks (and for clarity, this isn't the only place in the PR where this happened)

Reply:
done